tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -60,22 +74,35 @@ func (r *Report) writeToArtifactsDir() error {
 	return ioutil.WriteFile(outFilePath, contents, 0644)
 }
 
+// parseFlakyLog reads the latest flaky test report and returns filtered results based
+// on the function the caller passes in.
+func (c *Client) ParseFlakyLog(repo string, buildID int, f func(report Report, result *[]string)) ([]Report, []string, error) {
```

Comment:
this should not be called Parse. We are basically allowing callers to do anything they want with this. Any reason we are accepting a func as an arg, if we just want to parse?